### PR TITLE
[DTS] Only delete types of data that are being transferred

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

Update the import and transfer commands to generate the correct restore object

TODO:

- [ ] add more tests
- [ ] ~~refactor restore logic so it's modular and independent from local-source~~ I'm skipping this one because it would be a big structural change that might break things*

* I think we first need a discussion about if it would make more sense to make local-source a wrapper for a local-restore-source rather than local-source using strategies, since the strategies are/will be so tightly integrated it may be difficult to abstract them
 
### Why is it needed?

The initial functionality of the restore strategy causing a global reset of Strapi data caused too many users to be confused. 

The updated functionality of only touching data included in a transfer matches user expectations to not touch files that are excluded in a transfer or import.

### How to test it?

Using exclude or only with transfer or import should now only delete the files that are included in the process.

### Related issue(s)/PR(s)

Closes https://github.com/strapi/strapi/issues/17572

Completes DX-862, DX-863, DX-864

### CODE REVIEW

File: packages/core/data-transfer/src/engine/index.ts [hunk 1]
```diff
@@ -903,6 +911,9 @@ class TransferEngine<
 
   async transferConfiguration(): Promise<void> {
     const stage: TransferStage = 'configuration';
+    if (this.shouldSkipStage(stage)) {
```
[the remark this answers is not in the document]
Comment (innerdvations) [reply packages/core/data-transfer/src/engine/index.ts:914]: I think I did this to prevent it from being called even directly. I think it works for now. In the future, I can potentially see a use case for the transfer engine skipping it, but allowing it to be called manually, and we can update it then to allow in that case.

File: packages/core/data-transfer/src/strapi/providers/local-destination/index.ts [hunk 2]
```diff
@@ -229,23 +289,30 @@ class LocalStrapiDestinationProvider implements IDestinationProvider {
   async createAssetsWriteStream(): Promise<Writable> {
     assertValidStrapi(this.strapi, 'Not able to stream Assets');
 
+    if (!this.#areAssetsIncluded()) {
+      throw new ProviderTransferError('Attempting to transfer assets when they are not included');
+    }
+
     const removeAssetsBackup = this.#removeAssetsBackup.bind(this);
-    const strapi = this.strapi;
+    const thisStrapi = this.strapi;
```
Comment (Bassel17) [inline packages/core/data-transfer/src/strapi/providers/local-destination/index.ts:297]: not sure why this change was implemented, I almost confused this with as a `this.strapi` typo

File: packages/core/data-transfer/src/strapi/providers/local-destination/index.ts [hunk 3]
```diff
@@ -54,10 +58,40 @@ class LocalStrapiDestinationProvider implements IDestinationProvider {
   async bootstrap(): Promise<void> {
     this.#validateOptions();
     this.strapi = await this.options.getStrapi();
+    if (!this.strapi) {
+      throw new ProviderInitializationError('Could not access local strapi');
+    }
+
+    // uploading to a non-local file provider without included entities is useless (and potentially problematic) because we don't have the links, so don't allow it
+    if (
+      this.strapi.config.get('plugin.upload').provider !== 'local' &&
```
Comment (Bassel17) [inline packages/core/data-transfer/src/strapi/providers/local-destination/index.ts:67]: we can also not upload to local IMO, it would be better to start disconnecting DTS from upload providers specific logic, and also it's more consistent

**Review by Bassel17 — CHANGES_REQUESTED** [review]
Overall looks good, tested it as well

**Review by Bassel17 — CHANGES_REQUESTED** [review]
Overall looks good, tested it as well

**Review by christiancp100 — APPROVED** [review]
LGTM

**Review by derrickmehaffy — APPROVED** [review]
